### PR TITLE
Amount paid component

### PR DESCRIPTION
**Pull request by sauravmaiti22**
For the progress bar I had to use style instead of className because tailwind flex-grow-[] does not take decimal value.

### CODE REVIEW

**Review by Publicker — CHANGES_REQUESTED** [review]
Everything seems to be in order, with just a couple of tiny tweaks I think.

In the Figma design you can see that the numbers display "financial figures" and, despite using the same font, in the Storybook we can see that the numbers appear different. This is due to something called "font features". For the Inter font, these features are 'cv01', 'cv02', 'cv03', 'cv04'. We previously implemented this with CSS. However, Tailwind now offers this feature (refer to this post: https://tailwindcss.com/blog/tailwindcss-v3-3#configure-font-variation-settings-for-custom-font-families), so you can set it up there!

> If you want to know more about this font features for the Inter font family, you can see kind of the "documentation" over here: https://rsms.me/inter/

Apart from this, I noticed an issue when the paid amount exceeds the total amount. The line doesn't seem to respond correctly. My suggestion would be to keep the line at full length in these cases, as it's plausible that an overpayment could occur with a payment request.

**Review by pabloi-nf — CHANGES_REQUESTED** [review]
- The amount font-size needs to be 32px.
- Check the Inter font we are using. It seems to be different than the one using in the Figma (from Google Fonts).
- The line representing the percentage paid needs to have rounded edges.
- If there is an overpaid, the outstanding amount has to be 0. 
image

**Comment by sauravmaiti22** [comment]
> Everything seems to be in order, with just a couple of tiny tweaks I think.
> 
> In the Figma design you can see that the numbers display "financial figures" and, despite using the same font, in the Storybook we can see that the numbers appear different. This is due to something called "font features". For the Inter font, these features are 'cv01', 'cv02', 'cv03', 'cv04'. We previously implemented this with CSS. However, Tailwind now offers this feature (refer to this post: https://tailwindcss.com/blog/tailwindcss-v3-3#configure-font-variation-settings-for-custom-font-families), so you can set it up there!
> 
> > If you want to know more about this font features for the Inter font family, you can see kind of the "documentation" over here: https://rsms.me/inter/
> 
> Apart from this, I noticed an issue when the paid amount exceeds the total amount. The line doesn't seem to respond correctly. My suggestion would be to keep the line at full length in these cases, as it's plausible that an overpayment could occur with a payment request.

Thanks @Publicker. This was very useful. I did notice the difference in font but I thought I was hallucinating 😄 . I have made the changes. Please have a look.

**Comment by Publicker** [comment]
> > Everything seems to be in order, with just a couple of tiny tweaks I think.
> > In the Figma design you can see that the numbers display "financial figures" and, despite using the same font, in the Storybook we can see that the numbers appear different. This is due to something called "font features". For the Inter font, these features are 'cv01', 'cv02', 'cv03', 'cv04'. We previously implemented this with CSS. However, Tailwind now offers this feature (refer to this post: https://tailwindcss.com/blog/tailwindcss-v3-3#configure-font-variation-settings-for-custom-font-families), so you can set it up there!
> > > If you want to know more about this font features for the Inter font family, you can see kind of the "documentation" over here: https://rsms.me/inter/
> > 
> > 
> > Apart from this, I noticed an issue when the paid amount exceeds the total amount. The line doesn't seem to respond correctly. My suggestion would be to keep the line at full length in these cases, as it's plausible that an overpayment could occur with a payment request.
> 
> Thanks @Publicker. This was very useful. I did notice the difference in font but I thought I was hallucinating 😄 . I have made the changes. Please have a look.

@sauravmaiti22 Actually, disregard the font comment for now because something odd's happening there. Apparently the font works fine locally but it doesn't in remote, @pabloi-nf and myself are investigating.

**Comment by sauravmaiti22** [comment]
> * The amount font-size needs to be 32px.
> * Check the Inter font we are using. It seems to be different than the one using in the Figma (from Google Fonts).
> * The line representing the percentage paid needs to have rounded edges.
> * If there is an overpaid, the outstanding amount has to be 0.
> 
> image

Yes 

> 

I have made the changes. Thanks @pabloi-nf

File: src/components/ui/AmountPaid/AmountPaid.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+          <div className="bg-[#EDF2F7] rounded-r-sm" style={{ flexGrow: `${100 - percentagePaid}` }}></div>
+        </div>
+        {amountPaid > 0 && (
+          <span className="text-sm leading-[1.063rem] text-[#73808C] font-medium mt-2">
+            {currency == Currency.GBP ? '£' : '€'}
+            <span className="tabular-nums">{outstandingAmount.toFixed(2)}</span> outstanding
+          </span>
+        )}
+      </div>
+    </>
```
Comment (Publicker) [inline src/components/ui/AmountPaid/AmountPaid.tsx:36]: Just noticed a few more things.

1. Fragment (`<> </>`) is not needed as component returns only one element.
2. Substitution of classes should be done by using the `classNames` modules. [Here](https://github.com/nofrixion/nofrixion.webcomponents/blob/develop/src/components/ui/CreatePaymentRequestPage/CreatePaymentRequestPage.tsx#L409-L411)'s an example.
3. All HEX colours in here are actually available through names as they were added in the tailwind.config file.

Here are a few examples of classes to change:
- From `text-[#73808C]` to `text-greyText`
- From `text-[#73808C]` to `text-defaultText`
- From `bg-[#40BFBF]` to `bg-positiveActionBackground`
- and so on...

**Review by Publicker — CHANGES_REQUESTED** [review]

File: src/components/ui/AmountPaid/AmountPaid.tsx (hunk 1 above)
Comment (sauravmaiti22) [reply src/components/ui/AmountPaid/AmountPaid.tsx:36]: > Just noticed a few more things.
> 
> 1. Fragment (`<> </>`) is not needed as component returns only one element.
> 2. Substitution of classes should be done by using the `classNames` modules. [Here](https://github.com/nofrixion/nofrixion.webcomponents/blob/develop/src/components/ui/CreatePaymentRequestPage/CreatePaymentRequestPage.tsx#L409-L411)'s an example.
> 3. All HEX colours in here are actually available through names as they were added in the tailwind.config file.
> 
> Here are a few examples of classes to change:
> 
> * From `text-[#73808C]` to `text-greyText`
> * From `text-[#73808C]` to `text-defaultText`
> * From `bg-[#40BFBF]` to `bg-positiveActionBackground`
> * and so on...

Oh thanks @Publicker. Will make these changes.

**Review by Publicker — APPROVED** [review]

**Review by Publicker — APPROVED** [review]

**Review by pabloi-nf — APPROVED** [review]
Besides the font issue everything else looks good

**Pull request merged**